### DISCUSSION TITLE
Failed to build container in codespaces

### DISCUSSION THREAD

**Discussion by hiden52** [community/community#24992] [Codespaces]
<p>I used codespace until yesterday. I installed snapd for install heroku, but snap doesn’t worked. So I removed codespace and created a new codespace now. But I can’t create  a codespace.</p>
<pre><code class="lang-auto">$ Waiting for output...

$ git -C "/var/lib/docker/vsonlinemount/workspace" clone --depth 1 https://github.com/hiden52/LearnNodejs "/var/lib/docker/vsonlinemount/workspace/LearnNodejs"
Cloning into '/var/lib/docker/vsonlinemount/workspace/LearnNodejs'...
git process exited with exit code 0
$ git -C "/var/lib/docker/vsonlinemount/workspace/LearnNodejs" config --local remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
git process exited with exit code 0
$ git fetch --unshallow --tags

Starting devcontainer.json validation
devcontainer.json is valid

$ docker build -t cloudenvimage -f /var/lib/docker/vsonlinemount/workspace/LearnNodejs/.devcontainer/Dockerfile --build-arg "VARIANT=14" /var/lib/docker/vsonlinemount/workspace/LearnNodejs/.devcontainer
docker process exited with exit code 0
Sending build context to Docker daemon  4.096kB
Successfully built 148e10c9e712

Step 1/2 : ARG VARIANT="14-buster"
 ---&gt; 148e10c9e712
Step 2/2 : FROM mcr.microsoft.com/vscode/devcontainers/javascript-node:0-${VARIANT}
$ docker create --restart=no -v /root/.codespaces/shared:/home/node/.codespaces/shared -v /.codespaces/agent/mount:/.codespaces/bin -v /var/lib/docker/vsonlinemount/workspace:/home/node/workspace -v /var/lib/docker/vsonlinemount/.persistedshare:/home/node/.codespaces/.persistedshare -v /tmp/CoreFxPipe_VSLS:/tmp/CoreFxPipe_VSLS --name codespaces_a1c151 --hostname codespaces_a1c151 --add-host codespaces_a1c151:127.0.0.1 --label Type=codespaces --label StartCLI=False --label User=node --label ContainerVersion=8 --init -v /mnt/containerTmp:/tmp --entrypoint sh --network host -e "CODESPACES=true" cloudenvimage -c "while sleep 1000; do :; done"
47831f2f8c22d24a99beea64086b6d7c2b1004b04ce46f23f185aaf62378dd53
docker process exited with exit code 0
$ docker start codespaces_a1c151
codespaces_a1c151
docker process exited with exit code 0
$ /.codespaces/bin/Docker/linux-prereqs.sh true false false true

GitHub Codespaces Linux Dependency Installer

#See https://aka.ms/vsls-docs/linux-prerequisites

GitHub Codespaces requires a number of prerequisites that this script
will attempt to install for you. This process requires admin / root access.

(*) Detected Debian / Ubuntu
-e 
(*) Updating package lists...
git process exited with exit code 0
Get:1 http://deb.debian.org/debian buster InRelease [121 kB]
Get:2 http://security.debian.org/debian-security buster/updates InRelease [65.4 kB]
Get:3 http://deb.debian.org/debian buster-updates InRelease [51.9 kB]
Get:4 https://dl.yarnpkg.com/debian stable InRelease [17.1 kB]
Get:5 http://security.debian.org/debian-security buster/updates/main amd64 Packages [266 kB]
Get:6 http://deb.debian.org/debian buster/main amd64 Packages [7907 kB]
Get:7 http://deb.debian.org/debian buster-updates/main amd64 Packages [7848 B]
  The following signatures were invalid: EXPKEYSIG 23E7166788B63E1E Yarn Packaging &lt;yarn@dan.cx&gt;
Err:4 https://dl.yarnpkg.com/debian stable InRelease
Reading package lists...
(!) Failed to update list of available packages!
E: The repository 'https://dl.yarnpkg.com/debian stable InRelease' is not signed.
W: GPG error: https://dl.yarnpkg.com/debian stable InRelease: The following signatures were invalid: EXPKEYSIG 23E7166788B63E1E Yarn Packaging &lt;yarn@dan.cx&gt;
-e 
Press enter to dismiss this message
/.codespaces/bin/Docker/linux-prereqs.sh: 42: read: arg count
docker process exited with exit code 1
$ docker logs codespaces_a1c151
docker process exited with exit code 0
$ cat /etc/issue

Debian GNU/Linux 10 \n \l
cat process exited with exit code 0
The Codespaces Agent failed to start in the custom container.
</code></pre>

**Comment by jkeech** [comment, marked as answer]
<p><a class="mention" href="/hiden52">@hiden52</a>, we pushed out rebuilt images that have a newer GPG key embedded, so you shouldn’t run into this issue anymore. Let us know if you continue to see this error on new codespace creations.</p>
